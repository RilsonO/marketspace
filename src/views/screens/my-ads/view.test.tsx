import { act, fireEvent, render } from '@testing-library/react-native';
import { MyAds } from './view';
import { THEME } from '../../../theme';
import { MyAdsViewModel, useMyAdsViewModel } from './view-model';
import { FlatList, NativeBaseProvider } from 'native-base';
import { IProduct } from 'src/interfaces/product.interface';
import { UserModel } from 'src/models/user.model';
import { toMaskedPrice } from '@utils/Masks.util';
import { Plus } from 'phosphor-react-native';

jest.mock('./view-model');
const mockedNavigate = jest.fn();
jest.mock('@react-navigation/native', () => {
  const actualNav = jest.requireActual('@react-navigation/native');
  return {
    ...actualNav,
    useNavigation: () => ({
      navigate: mockedNavigate,
    }),
    useFocusEffect: jest.fn(),
  };
});

type Filter = 'Todos' | 'Ativos' | 'Inativos';

interface CreateMockMyAdsView {
  user: UserModel;
  products: IProduct[];
  filter: Filter;
  filterIsOpened: boolean;
}

const createMockMyAdsViewModel = ({
  user,
  products,
  filter,
  filterIsOpened,
}: CreateMockMyAdsView): MyAdsViewModel => {
  return {
    user,
    products,
    filter,
    filterIsOpened,
    productsUpdate: jest.fn(),
    handleOpenCreateAd: jest.fn(),
    handleFilterIsOpened: jest.fn(),
    handleFilter: jest.fn(),
  };
};

interface FactoryProps {
  user?: UserModel;
  products?: IProduct[];
  filter?: Filter;
  filterIsOpened?: boolean;
}

const defaultMockedUser = {
  name: 'John Doe',
  avatar: 'user_avatar.jpg',
  products: [],
} as unknown;

function makeSut({
  user = defaultMockedUser as UserModel,
  products = [],
  filter = 'Todos',
  filterIsOpened = false,
}: FactoryProps) {
  const mockMyAdsViewModel = createMockMyAdsViewModel({
    user,
    products,
    filter,
    filterIsOpened,
  });
  (useMyAdsViewModel as jest.Mock).mockReturnValueOnce(mockMyAdsViewModel);
  const inset = {
    frame: { x: 0, y: 0, width: 0, height: 0 },
    insets: { top: 0, left: 0, right: 0, bottom: 0 },
  };

  const sut = render(
    <NativeBaseProvider theme={THEME} initialWindowMetrics={inset}>
      <MyAds />
    </NativeBaseProvider>
  );

  return { ...sut, ...mockMyAdsViewModel };
}

describe('MyAds view', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render correctly', () => {
    const { toJSON, getByText } = makeSut({});
    const title = getByText('Meus anúncios');
    const countAds = getByText('0 anúncio');
    const filterTitle = getByText('Todos');
    expect(title).toBeTruthy();
    expect(countAds).toBeTruthy();
    expect(filterTitle).toBeTruthy();

    expect(toJSON()).toMatchSnapshot();
  });

  it('should show user products length correctly', () => {
    const mockedProducts: IProduct[] = [
      {
        id: '1',
        name: 'Product 1',
        description: 'Description for Product 1',
        is_new: true,
        price: 100,
        accept_trade: false,
        payment_methods: [],
        product_images: [],
        user: {
          id: '',
          avatar: '',
          name: '',
          email: '',
          tel: '',
        },
      },
      {
        id: '2',
        name: 'Product 2',
        description: 'Description for Product 2',
        is_new: true,
        price: 200,
        accept_trade: true,
        payment_methods: [],
        product_images: [],
        user: {
          id: '',
          avatar: '',
          name: '',
          email: '',
          tel: '',
        },
      },
      {
        id: '3',
        name: 'Product 3',
        description: 'Description for Product 3',
        is_new: false,
        price: 300,
        accept_trade: true,
        payment_methods: [],
        product_images: [],
        user: {
          id: '',
          avatar: '',
          name: '',
          email: '',
          tel: '',
        },
      },
    ];

    const { getByText } = makeSut({
      products: mockedProducts,
    });

    const countAds = getByText('0 anúncio');

    expect(countAds).toBeTruthy();
  });

  it('should render correctly with filter equal `Ativos`', async () => {
    const mockedFilter = 'Ativos';

    const { toJSON, getByText } = makeSut({
      filter: mockedFilter,
    });

    const filterTitle = await getByText(mockedFilter);

    expect(filterTitle).toBeTruthy();
    expect(toJSON()).toMatchSnapshot();
  });

  it('should render correctly with filter equal `Inativos`', async () => {
    const mockedFilter = 'Inativos';

    const { toJSON, getByText } = makeSut({
      filter: mockedFilter,
    });

    const filterTitle = await getByText(mockedFilter);

    expect(filterTitle).toBeTruthy();

    expect(toJSON()).toMatchSnapshot();
  });

  it('should render products correctly', () => {
    const mockedProducts: IProduct[] = [
      {
        id: '1',
        name: 'Product 1',
        price: 100,
        is_new: true,
        accept_trade: false,
        description: 'Description 1',
        payment_methods: [],
        product_images: [],
        user: {
          id: '',
          avatar: '',
          name: '',
          email: '',
          tel: '',
        },
      },
      {
        id: '2',
        name: 'Product 2',
        price: 200,
        is_new: false,
        accept_trade: true,
        description: 'Description 2',
        payment_methods: [],
        product_images: [],
        user: {
          id: '',
          avatar: '',
          name: '',
          email: '',
          tel: '',
        },
      },
    ];

    const { queryByText } = makeSut({
      products: mockedProducts,
    });

    mockedProducts.forEach((product) => {
      expect(queryByText(product.name)).toBeTruthy();
      expect(
        queryByText(`R$${toMaskedPrice(String(product.price))}`)
      ).toBeTruthy();
    });
  });

  it('should render products in two columns', () => {
    const mockedProducts: IProduct[] = [
      {
        id: '1',
        name: 'Product 1',
        description: 'Description for Product 1',
        is_new: true,
        price: 100,
        accept_trade: false,
        payment_methods: [],
        product_images: [],
        user: {
          id: '',
          avatar: '',
          name: '',
          email: '',
          tel: '',
        },
      },
      {
        id: '2',
        name: 'Product 2',
        description: 'Description for Product 2',
        is_new: true,
        price: 200,
        accept_trade: true,
        payment_methods: [],
        product_images: [],
        user: {
          id: '',
          avatar: '',
          name: '',
          email: '',
          tel: '',
        },
      },
      {
        id: '3',
        name: 'Product 3',
        description: 'Description for Product 3',
        is_new: false,
        price: 300,
        accept_trade: true,
        payment_methods: [],
        product_images: [],
        user: {
          id: '',
          avatar: '',
          name: '',
          email: '',
          tel: '',
        },
      },
    ];

    const { UNSAFE_getByType } = makeSut({
      products: mockedProducts,
    });

    const flatList = UNSAFE_getByType(FlatList);
    expect(flatList.props.numColumns).toBe(2);
  });

  it('should call handleOpenCreateAd when plus button is pressed', async () => {
    const { UNSAFE_getByType, handleOpenCreateAd } = makeSut({});

    await act(() => {
      fireEvent.press(UNSAFE_getByType(Plus));
    });

    expect(handleOpenCreateAd).toHaveBeenCalled();
  });

  it('should call handleFilterIsOpened when filter button is pressed', async () => {
    const { getByTestId, handleFilterIsOpened } = makeSut({});

    const filterButton = getByTestId('filter-button');

    await act(() => {
      fireEvent.press(filterButton);
    });

    expect(handleFilterIsOpened).toHaveBeenCalledWith(true);

    const todosFilterButton = getByTestId('todos-filter-item');
    const ativosFilterButton = getByTestId('ativos-filter-item');
    const inativosFilterButton = getByTestId('inativos-filter-item');

    expect(todosFilterButton).toBeTruthy();
    expect(ativosFilterButton).toBeTruthy();
    expect(inativosFilterButton).toBeTruthy();

    await act(() => {
      fireEvent.press(todosFilterButton);
    });

    expect(handleFilterIsOpened).toHaveBeenCalledWith(false);
  });
});
