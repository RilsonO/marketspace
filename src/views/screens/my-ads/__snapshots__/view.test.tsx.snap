// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyAds view should render correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    dataSet={{}}
    style={
      {
        "flex": 1,
        "paddingLeft": 24,
        "paddingRight": 24,
      }
    }
  >
    <View
      dataSet={{}}
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
          "marginBottom": 16,
          "marginTop": 8,
        }
      }
    >
      <Text
        dataSet={{}}
        style={
          {
            "backgroundColor": undefined,
            "color": "#1A181B",
            "fontFamily": "Karla_700Bold",
            "fontSize": 20,
            "fontStyle": "normal",
            "fontWeight": "400",
            "letterSpacing": 0,
            "lineHeight": 30,
            "textDecorationLine": undefined,
          }
        }
      >
        Meus anúncios
      </Text>
      <View
        dataSet={{}}
        style={{}}
      />
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        dataSet={{}}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "position": "absolute",
            "right": 0,
          }
        }
      >
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight="24"
          bbWidth="24"
          color="#1A181B"
          focusable={false}
          height={24}
          meetOrSlice={0}
          minX={0}
          minY={0}
          size={24}
          style={
            [
              {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              [
                undefined,
                undefined,
                {},
              ],
              {
                "flex": 0,
                "height": 24,
                "width": 24,
              },
            ]
          }
          tintColor="#1A181B"
          vbHeight={256}
          vbWidth={256}
          width={24}
        >
          <RNSVGGroup>
            <RNSVGRect
              fill={null}
              height="256"
              propList={
                [
                  "fill",
                ]
              }
              width="256"
              x="0"
              y="0"
            />
            <RNSVGLine
              fill={null}
              propList={
                [
                  "fill",
                  "stroke",
                  "strokeWidth",
                  "strokeLinecap",
                  "strokeLinejoin",
                ]
              }
              stroke={
                {
                  "payload": 4279900187,
                  "type": 0,
                }
              }
              strokeLinecap={1}
              strokeLinejoin={1}
              strokeWidth="24"
              x1="40"
              x2="216"
              y1="128"
              y2="128"
            />
            <RNSVGLine
              fill={null}
              propList={
                [
                  "fill",
                  "stroke",
                  "strokeWidth",
                  "strokeLinecap",
                  "strokeLinejoin",
                ]
              }
              stroke={
                {
                  "payload": 4279900187,
                  "type": 0,
                }
              }
              strokeLinecap={1}
              strokeLinejoin={1}
              strokeWidth="24"
              x1="128"
              x2="128"
              y1="40"
              y2="216"
            />
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
    </View>
    <View
      dataSet={{}}
      style={{}}
    />
    <View
      dataSet={{}}
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        dataSet={{}}
        style={
          {
            "backgroundColor": undefined,
            "color": "#3E3A40",
            "fontFamily": "Karla_400Regular",
            "fontSize": 14,
            "fontStyle": "normal",
            "fontWeight": "400",
            "letterSpacing": 0,
            "lineHeight": 21,
            "textDecorationLine": undefined,
          }
        }
      >
        0
         anúncio
      </Text>
      <View
        dataSet={{}}
        style={{}}
      />
      <View
        accessibilityLabel="More options menu"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        aria-haspopup="menu"
        collapsable={false}
        dataSet={{}}
        focusable={true}
        nativeID="1"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDownCapture={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "borderColor": "#9F9BA1",
            "borderRadius": 6,
            "borderWidth": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingBottom": 8,
            "paddingLeft": 12,
            "paddingRight": 12,
            "paddingTop": 8,
            "width": 111,
          }
        }
        testID="filter-button"
      >
        <Text
          dataSet={{}}
          style={
            {
              "backgroundColor": undefined,
              "color": "#1A181B",
              "fontFamily": "Karla_400Regular",
              "fontSize": 14,
              "fontStyle": "normal",
              "fontWeight": "400",
              "letterSpacing": 0,
              "lineHeight": 21,
              "textDecorationLine": undefined,
            }
          }
        >
          Todos
        </Text>
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight="16"
          bbWidth="16"
          color="#5F5B62"
          focusable={false}
          height={16}
          meetOrSlice={0}
          minX={0}
          minY={0}
          size={16}
          style={
            [
              {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              [
                undefined,
                undefined,
                {},
              ],
              {
                "flex": 0,
                "height": 16,
                "width": 16,
              },
            ]
          }
          tintColor="#5F5B62"
          vbHeight={256}
          vbWidth={256}
          width={16}
        >
          <RNSVGGroup>
            <RNSVGRect
              fill={null}
              height="256"
              propList={
                [
                  "fill",
                ]
              }
              width="256"
              x="0"
              y="0"
            />
            <RNSVGPath
              d="M208 96 128 176 48 96"
              fill={null}
              propList={
                [
                  "fill",
                  "stroke",
                  "strokeWidth",
                  "strokeLinecap",
                  "strokeLinejoin",
                ]
              }
              stroke={
                {
                  "payload": 4284439394,
                  "type": 0,
                }
              }
              strokeLinecap={1}
              strokeLinejoin={1}
              strokeWidth="16"
            />
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
    </View>
    <View
      dataSet={{}}
      style={{}}
    />
    <RCTScrollView
      contentContainerStyle={
        [
          {},
          {
            "dataSet": {},
          },
          {},
        ]
      }
      data={[]}
      dataSet={{}}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      showsVerticalScrollIndicator={false}
      stickyHeaderIndices={[]}
      style={
        [
          {},
          {
            "marginTop": 15,
          },
        ]
      }
      viewabilityConfigCallbackPairs={[]}
    >
      <View />
    </RCTScrollView>
  </View>
</RNCSafeAreaProvider>
`;

exports[`MyAds view should render correctly with filter equal \`Ativos\` 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    dataSet={{}}
    style={
      {
        "flex": 1,
        "paddingLeft": 24,
        "paddingRight": 24,
      }
    }
  >
    <View
      dataSet={{}}
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
          "marginBottom": 16,
          "marginTop": 8,
        }
      }
    >
      <Text
        dataSet={{}}
        style={
          {
            "backgroundColor": undefined,
            "color": "#1A181B",
            "fontFamily": "Karla_700Bold",
            "fontSize": 20,
            "fontStyle": "normal",
            "fontWeight": "400",
            "letterSpacing": 0,
            "lineHeight": 30,
            "textDecorationLine": undefined,
          }
        }
      >
        Meus anúncios
      </Text>
      <View
        dataSet={{}}
        style={{}}
      />
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        dataSet={{}}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "position": "absolute",
            "right": 0,
          }
        }
      >
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight="24"
          bbWidth="24"
          color="#1A181B"
          focusable={false}
          height={24}
          meetOrSlice={0}
          minX={0}
          minY={0}
          size={24}
          style={
            [
              {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              [
                undefined,
                undefined,
                {},
              ],
              {
                "flex": 0,
                "height": 24,
                "width": 24,
              },
            ]
          }
          tintColor="#1A181B"
          vbHeight={256}
          vbWidth={256}
          width={24}
        >
          <RNSVGGroup>
            <RNSVGRect
              fill={null}
              height="256"
              propList={
                [
                  "fill",
                ]
              }
              width="256"
              x="0"
              y="0"
            />
            <RNSVGLine
              fill={null}
              propList={
                [
                  "fill",
                  "stroke",
                  "strokeWidth",
                  "strokeLinecap",
                  "strokeLinejoin",
                ]
              }
              stroke={
                {
                  "payload": 4279900187,
                  "type": 0,
                }
              }
              strokeLinecap={1}
              strokeLinejoin={1}
              strokeWidth="24"
              x1="40"
              x2="216"
              y1="128"
              y2="128"
            />
            <RNSVGLine
              fill={null}
              propList={
                [
                  "fill",
                  "stroke",
                  "strokeWidth",
                  "strokeLinecap",
                  "strokeLinejoin",
                ]
              }
              stroke={
                {
                  "payload": 4279900187,
                  "type": 0,
                }
              }
              strokeLinecap={1}
              strokeLinejoin={1}
              strokeWidth="24"
              x1="128"
              x2="128"
              y1="40"
              y2="216"
            />
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
    </View>
    <View
      dataSet={{}}
      style={{}}
    />
    <View
      dataSet={{}}
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        dataSet={{}}
        style={
          {
            "backgroundColor": undefined,
            "color": "#3E3A40",
            "fontFamily": "Karla_400Regular",
            "fontSize": 14,
            "fontStyle": "normal",
            "fontWeight": "400",
            "letterSpacing": 0,
            "lineHeight": 21,
            "textDecorationLine": undefined,
          }
        }
      >
        0
         anúncio
      </Text>
      <View
        dataSet={{}}
        style={{}}
      />
      <View
        accessibilityLabel="More options menu"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        aria-haspopup="menu"
        collapsable={false}
        dataSet={{}}
        focusable={true}
        nativeID="3"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDownCapture={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "borderColor": "#9F9BA1",
            "borderRadius": 6,
            "borderWidth": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingBottom": 8,
            "paddingLeft": 12,
            "paddingRight": 12,
            "paddingTop": 8,
            "width": 111,
          }
        }
        testID="filter-button"
      >
        <Text
          dataSet={{}}
          style={
            {
              "backgroundColor": undefined,
              "color": "#1A181B",
              "fontFamily": "Karla_400Regular",
              "fontSize": 14,
              "fontStyle": "normal",
              "fontWeight": "400",
              "letterSpacing": 0,
              "lineHeight": 21,
              "textDecorationLine": undefined,
            }
          }
        >
          Ativos
        </Text>
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight="16"
          bbWidth="16"
          color="#5F5B62"
          focusable={false}
          height={16}
          meetOrSlice={0}
          minX={0}
          minY={0}
          size={16}
          style={
            [
              {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              [
                undefined,
                undefined,
                {},
              ],
              {
                "flex": 0,
                "height": 16,
                "width": 16,
              },
            ]
          }
          tintColor="#5F5B62"
          vbHeight={256}
          vbWidth={256}
          width={16}
        >
          <RNSVGGroup>
            <RNSVGRect
              fill={null}
              height="256"
              propList={
                [
                  "fill",
                ]
              }
              width="256"
              x="0"
              y="0"
            />
            <RNSVGPath
              d="M208 96 128 176 48 96"
              fill={null}
              propList={
                [
                  "fill",
                  "stroke",
                  "strokeWidth",
                  "strokeLinecap",
                  "strokeLinejoin",
                ]
              }
              stroke={
                {
                  "payload": 4284439394,
                  "type": 0,
                }
              }
              strokeLinecap={1}
              strokeLinejoin={1}
              strokeWidth="16"
            />
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
    </View>
    <View
      dataSet={{}}
      style={{}}
    />
    <RCTScrollView
      contentContainerStyle={
        [
          {},
          {
            "dataSet": {},
          },
          {},
        ]
      }
      data={[]}
      dataSet={{}}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      showsVerticalScrollIndicator={false}
      stickyHeaderIndices={[]}
      style={
        [
          {},
          {
            "marginTop": 15,
          },
        ]
      }
      viewabilityConfigCallbackPairs={[]}
    >
      <View />
    </RCTScrollView>
  </View>
</RNCSafeAreaProvider>
`;

exports[`MyAds view should render correctly with filter equal \`Inativos\` 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    dataSet={{}}
    style={
      {
        "flex": 1,
        "paddingLeft": 24,
        "paddingRight": 24,
      }
    }
  >
    <View
      dataSet={{}}
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
          "marginBottom": 16,
          "marginTop": 8,
        }
      }
    >
      <Text
        dataSet={{}}
        style={
          {
            "backgroundColor": undefined,
            "color": "#1A181B",
            "fontFamily": "Karla_700Bold",
            "fontSize": 20,
            "fontStyle": "normal",
            "fontWeight": "400",
            "letterSpacing": 0,
            "lineHeight": 30,
            "textDecorationLine": undefined,
          }
        }
      >
        Meus anúncios
      </Text>
      <View
        dataSet={{}}
        style={{}}
      />
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        dataSet={{}}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "position": "absolute",
            "right": 0,
          }
        }
      >
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight="24"
          bbWidth="24"
          color="#1A181B"
          focusable={false}
          height={24}
          meetOrSlice={0}
          minX={0}
          minY={0}
          size={24}
          style={
            [
              {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              [
                undefined,
                undefined,
                {},
              ],
              {
                "flex": 0,
                "height": 24,
                "width": 24,
              },
            ]
          }
          tintColor="#1A181B"
          vbHeight={256}
          vbWidth={256}
          width={24}
        >
          <RNSVGGroup>
            <RNSVGRect
              fill={null}
              height="256"
              propList={
                [
                  "fill",
                ]
              }
              width="256"
              x="0"
              y="0"
            />
            <RNSVGLine
              fill={null}
              propList={
                [
                  "fill",
                  "stroke",
                  "strokeWidth",
                  "strokeLinecap",
                  "strokeLinejoin",
                ]
              }
              stroke={
                {
                  "payload": 4279900187,
                  "type": 0,
                }
              }
              strokeLinecap={1}
              strokeLinejoin={1}
              strokeWidth="24"
              x1="40"
              x2="216"
              y1="128"
              y2="128"
            />
            <RNSVGLine
              fill={null}
              propList={
                [
                  "fill",
                  "stroke",
                  "strokeWidth",
                  "strokeLinecap",
                  "strokeLinejoin",
                ]
              }
              stroke={
                {
                  "payload": 4279900187,
                  "type": 0,
                }
              }
              strokeLinecap={1}
              strokeLinejoin={1}
              strokeWidth="24"
              x1="128"
              x2="128"
              y1="40"
              y2="216"
            />
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
    </View>
    <View
      dataSet={{}}
      style={{}}
    />
    <View
      dataSet={{}}
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        dataSet={{}}
        style={
          {
            "backgroundColor": undefined,
            "color": "#3E3A40",
            "fontFamily": "Karla_400Regular",
            "fontSize": 14,
            "fontStyle": "normal",
            "fontWeight": "400",
            "letterSpacing": 0,
            "lineHeight": 21,
            "textDecorationLine": undefined,
          }
        }
      >
        0
         anúncio
      </Text>
      <View
        dataSet={{}}
        style={{}}
      />
      <View
        accessibilityLabel="More options menu"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        aria-haspopup="menu"
        collapsable={false}
        dataSet={{}}
        focusable={true}
        nativeID="4"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDownCapture={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "borderColor": "#9F9BA1",
            "borderRadius": 6,
            "borderWidth": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingBottom": 8,
            "paddingLeft": 12,
            "paddingRight": 12,
            "paddingTop": 8,
            "width": 111,
          }
        }
        testID="filter-button"
      >
        <Text
          dataSet={{}}
          style={
            {
              "backgroundColor": undefined,
              "color": "#1A181B",
              "fontFamily": "Karla_400Regular",
              "fontSize": 14,
              "fontStyle": "normal",
              "fontWeight": "400",
              "letterSpacing": 0,
              "lineHeight": 21,
              "textDecorationLine": undefined,
            }
          }
        >
          Inativos
        </Text>
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight="16"
          bbWidth="16"
          color="#5F5B62"
          focusable={false}
          height={16}
          meetOrSlice={0}
          minX={0}
          minY={0}
          size={16}
          style={
            [
              {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              [
                undefined,
                undefined,
                {},
              ],
              {
                "flex": 0,
                "height": 16,
                "width": 16,
              },
            ]
          }
          tintColor="#5F5B62"
          vbHeight={256}
          vbWidth={256}
          width={16}
        >
          <RNSVGGroup>
            <RNSVGRect
              fill={null}
              height="256"
              propList={
                [
                  "fill",
                ]
              }
              width="256"
              x="0"
              y="0"
            />
            <RNSVGPath
              d="M208 96 128 176 48 96"
              fill={null}
              propList={
                [
                  "fill",
                  "stroke",
                  "strokeWidth",
                  "strokeLinecap",
                  "strokeLinejoin",
                ]
              }
              stroke={
                {
                  "payload": 4284439394,
                  "type": 0,
                }
              }
              strokeLinecap={1}
              strokeLinejoin={1}
              strokeWidth="16"
            />
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
    </View>
    <View
      dataSet={{}}
      style={{}}
    />
    <RCTScrollView
      contentContainerStyle={
        [
          {},
          {
            "dataSet": {},
          },
          {},
        ]
      }
      data={[]}
      dataSet={{}}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      showsVerticalScrollIndicator={false}
      stickyHeaderIndices={[]}
      style={
        [
          {},
          {
            "marginTop": 15,
          },
        ]
      }
      viewabilityConfigCallbackPairs={[]}
    >
      <View />
    </RCTScrollView>
  </View>
</RNCSafeAreaProvider>
`;
